libraries/Microsoft.Bot.Builder.Dialogs/DialogExtensions.cs

```diff
@@ -34,6 +36,61 @@ public static async Task RunAsync(this Dialog dialog, ITurnContext turnContext,
 
             var dialogContext = await dialogSet.CreateContextAsync(turnContext, cancellationToken).ConfigureAwait(false);
 
+            // drop the return value - the whole point of this function is that the caller doesn't care
+            await InternalRunAsync(turnContext, dialog.Id, dialogContext, cancellationToken).ConfigureAwait(false);
+        }
+
+        internal static async Task<DialogTurnResult> InternalRunAsync(ITurnContext turnContext, string dialogId, DialogContext dialogContext, CancellationToken cancellationToken)
+        {
+            // map TurnState into root dialog context.services
+            foreach (var service in turnContext.TurnState)
```

Comment:
Added this because this is what DialogManager does. But seriously is this necessary?